datafusion/core/src/physical_plan/joins/utils.rs

```diff
@@ -438,7 +438,7 @@ impl<T: 'static> OnceAsync<T> {
 }
 
 /// The shared future type used internally within [`OnceAsync`]
-type OnceFutPending<T> = Shared<BoxFuture<'static, Arc<Result<T>>>>;
+type OnceFutPending<T> = Shared<BoxFuture<'static, Arc<SharedResult<T>>>>;
```

Comment:
Now that `SharedResult` has an `Arc`, can we avoid the double-arc here?

Reply:
`OnceFut` is cloneable, while `SharedResult` wraps only `Err` into `Arc` to allow cloning the error itself 🤔 

I decided to declare it as `SharedResult<Arc<T>>` instead, so it won't be `Arc` inside `Arc`. It will be either result or error wrapped into `Arc`.

Take a look at the last commit.